js/compiler.ts

```diff
@@ -201,6 +268,17 @@ class Compiler implements ts.LanguageServiceHost, ts.FormatDiagnosticsHost {
     );
   }
 
+  /** Log TypeScript diagnostics to the console and exit */
+  private _logDiagnostics(diagnostics: ts.Diagnostic[]): never {
+    const errMsg = this._os.noColor
+      ? this._ts.formatDiagnostics(diagnostics, this)
+      : this._ts.formatDiagnosticsWithColorAndContext(diagnostics, this);
+
+    console.log(errMsg);
+    // All TypeScript errors are terminal for deno
+    return this._os.exit(1);
```

Comment:
I realize you're just moving this around, but can you add a TODO here ... something like this (feel free to modify)
```
// TODO The compiler isolate shouldn't exit, but rather errors should be 
// forwarded back to the controller, where an exit will be invoked.
```